src/ZoneServer/Skills/Handlers/Sapper/PunjiStake.cs

```diff
@@ -0,0 +1,279 @@
+		private async void ExplodeTrap(ICombatEntity caster, Skill skill, Mob trap, int effectId, CancellationTokenSource cancellationTokenSource)
+		{
+			var character = caster as Character;
+
+			cancellationTokenSource.Cancel();
+
+			Send.ZC_NORMAL.Skill_50(character, skill.Id, 1.5f);
+
+			await Task.Delay(TimeSpan.FromMilliseconds(150));
+
+			Send.ZC_NORMAL.PlayAnimationOnEffect_6D(character, effectId);
+
+			var splashArea = new Square(trap.Position.GetRelative(trap.Direction, -50), trap.Direction, 90, 40);
+
+			Debug.ShowShape(character.Map, splashArea, edgePoints: false);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				this.ExecuteAttack(caster, target, skill, trap);
+			}
+
+			await Task.Delay(TimeSpan.FromMilliseconds(800));
+
+			Send.ZC_NORMAL.PlayAnimationOnEffect_7D(character, skill.Id);
+
+			await Task.Delay(TimeSpan.FromSeconds(2));
+
+			Send.ZC_DEAD(trap, trap.Position);
+			caster.Map.RemoveMonster(trap);
+
+			Send.ZC_NORMAL.GroundEffect_59(character, "punji_stake", skill.Id, trap.Position, effectId, false);
+
+		}
+
+		/// <summary>
+		/// Execute the trap attack after the explosion
+		/// </summary>
+		/// <param name="caster"></param>
+		/// <param name="target"></param>
+		/// <param name="skill"></param>
+		/// <param name="trap"></param>
+		private async void ExecuteAttack(ICombatEntity caster, ICombatEntity target, Skill skill, Mob trap)
+		{
+			var character = caster as Character;
+
+			var skillHitResult = SCR_SkillHit(caster, target, skill);
+
+			// Ability - Poonji Stake: Instance
+			// Poonji Steak is immediately installed and damage is reduced by 50%{nl}*
+			// When attacked by Punji Steak, the movement speed is fixed at 10 for 5 seconds,
+			// and the damage received from the caster is increased by 30%
+			if (character.Abilities.IsActive(AbilityId.Sapper42))
+			{
+				var mspd = target.Properties.GetFloat(PropertyName.MSPD);
+				skillHitResult.Damage *= 0.5f;
+				this.ApplySlow(target, mspd);
+				return;
+			}
+
+			var knockBackDistance = 110;
+			var knockBackPos = target.Position.GetRelative(trap.Direction, knockBackDistance);
+			var angle = target.GetDirection(knockBackPos).DegreeAngle;
+
+			Send.ZC_KNOCKDOWN_INFO(character, target, target.Position, knockBackPos, angle);
+
+			target.Position = knockBackPos;
+
+			target.TakeDamage(skillHitResult.Damage, caster);
+
+			var hit = new HitInfo(caster, target, skill, skillHitResult);
+
+			for (int i = 0; i < 4; i++)
+			{
+				Send.ZC_HIT_INFO(caster, target, skill, hit);
+				await Task.Delay(100);
+			}
+		}
+
+		/// <summary>
+		/// Apply Slow to the targets and restore it after certain time
+		/// </summary>
+		/// <param name="target"></param>
+		/// <param name="oldValue"></param>
+		private async void ApplySlow(ICombatEntity target, float oldValue)
+		{
+			target.Properties.SetFloat(PropertyName.MSPD, 10);
```

Comment:
You can't change MSPD directly because it's a calculated property. But even if you could change it, that would be a pretty bad idea to do from a skill like this, because if the target were to log out in between setting and resetting the value, they'd get stuck with the lower MSPD when it's saved to the db.

Reply:
Worked around it: Now it applies 70% move speed reduction.